controllers/helloworld_controller.go

```diff
@@ -72,24 +63,72 @@ func (r *HelloworldReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	stopReconcile, result, err := r.validateOrCreateSvcFor(helloworld, ctx, req)
+
+	if stopReconcile {
+		return result, err
+	}
+
+	stopReconcile, result, err = r.validateOrCreatePodsFor(helloworld, ctx, req)
+
+	if stopReconcile {
+		return result, err
+	}
+
+	Logging(req, "Finished succesfully")
+
+	return ctrl.Result{Requeue: false}, nil
+}
+
+func (r *HelloworldReconciler) validateOrCreateSvcFor(helloworld *v1alpha1.Helloworld, ctx context.Context, req ctrl.Request) (bool, ctrl.Result, error) {
 	svc := &v1.Service{}
 	svcNamespacedName := types.NamespacedName{Namespace: helloworld.Namespace, Name: helloworld.Name + "-service"}
+	stopReconcile := true
 
 	if err := r.Get(ctx, svcNamespacedName, svc); err != nil {
 		if errors.IsNotFound(err) {
 			Logging(req, "Creating a service")
-			r.Create(ctx, r.svcForHelloworld(helloworld))
-			return ctrl.Result{Requeue: true}, nil
+			r.Create(ctx, r.defineSvcFor(helloworld))
```

Comment:
always always always check the error of functions that return error.
```suggestion
			err := r.Create(ctx, r.defineSvcFor(helloworld))
			if err != nil {
			    return ...... err
			}
```